posthog/models/user.py

```diff
@@ -176,30 +176,32 @@ def teams(self):
         All teams the user has access to on any organization, taking into account project based permissioning
         """
         teams = Team.objects.filter(organization__members=self)
-        if Organization.objects.filter(
-            members=self,
-            available_features__contains=[AvailableFeature.PROJECT_BASED_PERMISSIONING],
-        ).exists():
-            try:
-                from ee.models import ExplicitTeamMembership
-            except ImportError:
-                pass
-            else:
-                available_private_project_ids = ExplicitTeamMembership.objects.filter(
-                    Q(parent_membership__user=self)
-                ).values_list("team_id", flat=True)
-                organizations_where_user_is_admin = OrganizationMembership.objects.filter(
-                    user=self, level__gte=OrganizationMembership.Level.ADMIN
-                ).values_list("organization_id", flat=True)
-                # If project access control IS applicable, make sure
-                # - project doesn't have access control OR
-                # - the user has explicit access OR
-                # - the user is Admin or owner
-                teams = teams.filter(
-                    Q(access_control=False)
-                    | Q(pk__in=available_private_project_ids)
-                    | Q(organization__pk__in=organizations_where_user_is_admin)
-                )
+        org_available_product_features = Organization.objects.filter(members=self).values_list(
+            "available_product_features", flat=True
+        )
+        if org_available_product_features[0]:
```

Comment:
this is not ok, will fail when say this list returns an empty queryset, which while maybe a ultra rare case, our code should be defensive.

You should use .len() or .exists() in conjunction with checking the value of the first item

```suggestion
        if org_available_product_features.exists() and org_available_product_features[0]:
```

Reply:
or maybe even better, use `.first()`